<!-- RGB-D Monocual Visual Odometry -->

<launch>
  <arg name="PATH_TO_KEYFRAMES" default="$(env HOME)/ros/Maps/3D/CCNY Office 640/Keyframes" />
  
  <node pkg="ccny_rgbd" type="keyframes_reshaper" name="keyframes_reshaper" 
    output="screen">

    # Keyframes location:
    <param name="apps/keyframes_reshaper/path_to_keyframes" value="$(arg PATH_TO_KEYFRAMES)" />
    <param name="apps/keyframes_reshaper/first_keyframe_number" value="0" /> 
    <param name="apps/keyframes_reshaper/last_keyframe_number" value="1" /> 
    
    # Filter values:
    <param name="apps/keyframes_reshaper/vgf_res" value="0.005" />
    <param name="apps/keyframes_reshaper/neighbor_max_proximity" value="0.008" /> 
    <param name="apps/keyframes_reshaper/smoothing_res" value="0.0" /> 

    # PCD file location
    <param name="apps/keyframes_reshaper/output_filename" value="cloud_reshaped"/> 
  </node>

</launch>